batch_job/src/batch_job_amazon_batch.c

```diff
@@ -0,0 +1,567 @@
+	clean_str_array(files_split_num,&files_split);
+	return success;
+}
+
+
+static struct internal_amazon_batch_amazon_ids initialize(struct batch_queue* q){
+	if(initialized){
+		return initialized_data;
+	}
+	char* config_file = hash_table_lookup(q->options,"amazon-batch-config");
+	if(!config_file) {
+		fatal("No amazon config file passed!");
+	}
+	
+	struct jx* config = jx_parse_file(config_file);
+	
+	
+	initialized = 1;
+	instID = time(NULL);
+	queue_name = string_format("%i_ccl_amazon_batch_queue",instID);//should be unique
+	done_jobs = itable_create(0);//default size
+	amazon_job_ids = itable_create(0);
+	done_files = itable_create(0);
+	submitted_files = hash_table_create(0,0);
+		
+
+	char* amazon_ami = hash_table_lookup(q->options,"amazon-batch-img");
+	if(amazon_ami == NULL) {
+		fatal("No image id passed. Please pass file containing ami image id using --amazon-batch-img flag");
+	}
+
+	char* aws_access_key_id     = (char*)jx_lookup_string(config, "aws_id");
+	char* aws_secret_access_key = (char*)jx_lookup_string(config, "aws_key");
+	char* aws_region            = (char*)jx_lookup_string(config,"aws_reg");
+	bucket_name                 = (char*)jx_lookup_string(config,"bucket");
+	vpc                         = (char*)jx_lookup_string(config,"vpc");
+	sec_group                   = (char*)jx_lookup_string(config,"sec_group");
+	queue_name                  = (char*)jx_lookup_string(config,"queue_name");
+	compute_env_name            = (char*)jx_lookup_string(config,"env_name");
+	subnet                      = (char*)jx_lookup_string(config,"subnet");	
+
+	if(!aws_access_key_id)
+		fatal("credentials file %s does not contain aws_id",config_file);
+	if(!aws_secret_access_key)
+		fatal("credentials file %s does not contain aws_key",config_file);
+	if(!aws_region)
+		fatal("credentials file %s does not contain aws_reg",config_file);
+	if(!bucket_name)
+		fatal("credentials file %s does not contain bucket",config_file);
+	if(!queue_name)
+		fatal("credentials file %s does not contain queue_name",config_file);
+	if(!compute_env_name)
+		fatal("credentials file %s does not contain env_name",config_file);
+	if(!vpc)
+		fatal("credentials file %s does not contain vpc",config_file);
+	if(!subnet)
+		fatal("credentials file %s does not contain subnet",config_file); 
+	
+	char* env_var = string_format("AWS_ACCESS_KEY_ID=%s AWS_SECRET_ACCESS_KEY=%s AWS_DEFAULT_REGION=%s ",aws_access_key_id,aws_secret_access_key,aws_region);
+	
+
+	initialized_data.aws_access_key_id = aws_access_key_id;
+	initialized_data.aws_secret_access_key = aws_secret_access_key;
+	initialized_data.aws_region=aws_region;
+	initialized_data.master_env_prefix = env_var;
+	return initialized_data;
+}
+
+static char* generate_s3_cp_cmds(char* files, char* src, char* dst){
+	char* env_var = initialized_data.master_env_prefix;
+	int i;
+	int files_split_num = 0;
+	char** files_split;
+	split_comma_list(files, &files_split_num, &files_split);
+
+	char* new_cmd=malloc(sizeof(char)*1);
```

Comment:
Why do you need to malloc here?

Reply:
By having a malloc here, it makes the logic inside the for loop cleaner without needing a special case for the first time it tries to append to the string.

Reply:
Also, testing with "," as the output file didn't cause an infinite loop